### PR TITLE
components: QueryState HOC

### PR DESCRIPTION
**Pull request by zzacharo**
- Adds a new HOC component `withQueryState` which enables an external component to read the state.
- NOTE: The connected component *must* be child of the `ReactSearchkit` component.

### CODE REVIEW

**Comment by zzacharo** [comment]
I also added a new component in the favour of showing how to use the HOC component. See screenshot and let me know if it is ok to keep it!
Screenshot 2019-08-06 at 22 03 29

**Comment by ntarocco** [comment]
> I also added a new component in the favour of showing how to use the HOC component. See screenshot and let me know if it is ok to keep it!
> Screenshot 2019-08-06 at 22 03 29

I think this should go into the doc and not into the demo, no?
It looks super ugly! :)

File: demo/src/index.js [hunk 1]
```diff
@@ -23,10 +23,23 @@ import {
...
   Error,
   ResultsLoader,
+  withQueryState,
 } from '@app/components';
 import { Results } from './Results';
 import { InvenioSearchApi } from '@app/api';
 
+class _QueryStateLogger extends Component {
+  render() {
+    console.log('Current query state', this.props);
```
Comment (ntarocco) [inline demo/src/index.js]: oups!

File: src/contrib/Serializers.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/*
+ * This file is part of React-SearchKit.
+ * Copyright (C) 2018 CERN.
```
Comment (ntarocco) [inline src/contrib/Serializers.js]: 2019

File: src/contrib/Serializers.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
+/*
```
Comment (ntarocco) [inline src/contrib/Serializers.js]: I think you are missing a subfolder `invenio` inside `contrib`, unless you had other ideas in mind!

File: src/components/HOC/withQueryState.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*
+ * This file is part of React-SearchKit.
+ * Copyright (C) 2018 CERN.
```
Comment (ntarocco) [inline src/components/HOC/withQueryState.js]: 2019

File: src/components/HOC/withQueryState.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+import React from 'react';
+import { connect } from '@app/store';
+
+export function withQueryState(Component) {
+  const WrappedComponent = ({ dispatch, ...props }) => <Component {...props} />;
+  const mapStateToProps = state => ({
+    currentQueryState: state.query,
+    totalResults: state.results.data.total,
```
Comment (ntarocco) [inline src/components/HOC/withQueryState.js]: do you need this? At this point, I would expose:
```
{
    currentQueryState: state.query,
    currentResultsState: state.results,
}
```
no?

**Pull request merged**